### DISCUSSION TITLE
Enable layer with modifiers active

### DISCUSSION THREAD

**Discussion by xauser** [nickcoutsos/keymap-editor#273] [Q&A]
Hello,

I'm looking for a way to have a key that switches to a layer and has two modifiers enabled. In TMK this is done with macro `ACTION_LAYER_MODS`.

I found this thread [ZMK Issue](https://github.com/zmkfirmware/zmk/issues/968) that says it's possible using macros (example [ZMK Doc](https://zmk.dev/docs/keymaps/behaviors/macros#layer--modifier)).

I tried my best to configure this in the keymap-editor macro section and assign this macro to a key on my keymap.

When I press 'save' the build fails with...

```
devicetree error: /keymap/default_layer: undefined node label 'lm'
CMake Error at /__w/Adv360-Pro-ZMK/Adv360-Pro-ZMK/zephyr/cmake/modules/dts.cmake:279 (message):
gen_defines.py failed with return code: 1
Call Stack (most recent call first):
/__w/Adv360-Pro-ZMK/Adv360-Pro-ZMK/zephyr/cmake/modules/zephyr_default.cmake:129 (include)
/__w/Adv360-Pro-ZMK/Adv360-Pro-ZMK/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
/__w/Adv360-Pro-ZMK/Adv360-Pro-ZMK/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
CMakeLists.txt:8 (find_package)
```

**Comment by nickcoutsos** [comment]
I think the `#ifndef VERSION_MACRO`/`#endif` is excluding it from the compile code. If you don't want to get rid of that altogether you can just manually move your macro after the `#endif` line.

**Comment by xauser** [comment]
I replaced the macro created by the editor with the one from the zmk doc and moved it below the `#endif` and that worked. It would be great to get this done with the editor some day.

**Reply by nickcoutsos** [reply]
It would but I can't overstate the complexity of trying to support the C preprocessor at the same level as devicetree. I go into some of the problems here: https://github.com/nickcoutsos/keymap-editor/wiki/Preprocessor-Support
